OWL2/PrintOWL2AS.hs

```diff
@@ -0,0 +1,608 @@
+        [map pretty axAnns, [pretty objPropExpr, pretty srcInd,
+            pretty targInd]])
+
+printDataPropertyAssertion :: AxiomAnnotations -> DataPropertyExpression
+    -> SourceIndividual -> TargetValue -> Doc
+printDataPropertyAssertion axAnns dataPropExpr srcInd targVal =
+    keyword dataPropertyAssertionS
+    <> sParens (hsep . concat $
+      [map pretty axAnns, [pretty dataPropExpr, pretty srcInd,
+        pretty targVal]])
+
+printNegativeDataPropertyAssertion :: AxiomAnnotations -> DataPropertyExpression
+    -> SourceIndividual -> TargetValue -> Doc
+printNegativeDataPropertyAssertion axAnns dataPropExpr srcInd targVal =
+    keyword negativeDataPropertyAssertionS
+    <> sParens (hsep . concat $
+        [map pretty axAnns, [pretty dataPropExpr, pretty srcInd,
+            pretty targVal]])
+
+-- | print AnnotationAxiom
+instance Pretty AnnotationAxiom where
+    pretty ax = case ax of 
+        AnnotationAssertion axAnns annProp annSub annValue
+            -> printAnnotationAssertion axAnns annProp annSub annValue
+        SubAnnotationPropertyOf axAnns subAnnProp supAnnProp
+            -> printSubAnnotationPropertyOf axAnns subAnnProp supAnnProp
+        AnnotationPropertyDomain axAnns annProp iri
+            -> printAnnotationPropertyDomain axAnns annProp iri
+        AnnotationPropertyRange axAnns annProp iri
+            -> printAnnotationPropertyRange axAnns annProp iri
+
+printAnnotationAssertion :: AxiomAnnotations -> AnnotationProperty
+    -> AnnotationSubject ->  AnnotationValue -> Doc
+printAnnotationAssertion axAnns annProp annSub annValue =
+    keyword annotationAssertionS
+    <> sParens (hsep . concat $
+        [map pretty axAnns, [pretty annProp, pretty annSub,
+            pretty annValue]])
+
+printSubAnnotationPropertyOf :: AxiomAnnotations -> SubAnnotationProperty
+    -> SuperAnnotationProperty -> Doc
+printSubAnnotationPropertyOf axAnns subAnnProp supAnnProp =
+    keyword subAnnotationPropertyOfS
+    <> sParens (hsep . concat $
+        [map pretty axAnns, [pretty subAnnProp, pretty supAnnProp]])
+
+printAnnotationPropertyDomain :: AxiomAnnotations -> AnnotationProperty
+    -> IRI -> Doc
+printAnnotationPropertyDomain axAnns annProp iri =
+    keyword annotationPropertyDomainS
+    <> sParens (hsep . concat $
+        [map pretty axAnns, [pretty annProp, pretty iri]])
+
+printAnnotationPropertyRange :: AxiomAnnotations -> AnnotationProperty
+    -> IRI -> Doc
+printAnnotationPropertyRange axAnns annProp iri =
+    keyword annotationPropertyRangeS
+    <> sParens (hsep . concat $
+        [map pretty axAnns, [pretty annProp, pretty iri]])
+
+-- | print Root
+instance Pretty PrefixDeclaration where
+    pretty (PrefixDeclaration prName iri) =
+        keyword "Prefix"
+        <> sParens ((text prName) <> (text "=") <> pretty iri)
+
+instance Pretty Ontology where
+    pretty (Ontology mOnt mVerIri dImpDocs ontAnns axioms) =
+        keyword "Ontology"
+        <> sParens (hsep . concat $ 
```

Comment:
Use `vsep` instead of `hsep`, otherwise you get the whole ontology on one very long line.